contracts/SimpleOperations.sol

```diff
@@ -45,121 +46,258 @@ contract SimpleOperations is Operations {
 	// Functions for client owners
 
-	function setClientOwner(address _newOwner) public only_client_owner {
-		var newClient = clientOwner[msg.sender];
-		clientOwner[msg.sender] = bytes32(0);
+	function setClientOwner(address _newOwner)
+		public
+		onlyClientOwner
+		notClientOwner(_newOwner)
+	{
+		bytes32 newClient = clientOwner[msg.sender];
+		delete clientOwner[msg.sender];
+
 		clientOwner[_newOwner] = newClient;
 		client[newClient].owner = _newOwner;
-		ClientOwnerChanged(newClient, msg.sender, _newOwner);
 	}
 
-	function addRelease(bytes32 _release, uint32 _forkBlock, uint8 _track, uint24 _semver, bool _critical) public only_client_owner {
-		var newClient = clientOwner[msg.sender];
-		client[newClient].release[_release] = Release(_forkBlock, _track, _semver, _critical);
+	/// Adds a release which is identified by `_release`. When a release is succesfully added, a
+	/// `ReleaseAdded` event is emitted. The same release can be added multiple times (with
+	/// different parameters), therefore the `ReleaseAdded` event can be emitted multiple times for
+	/// the same release.
+	function addRelease(
+		bytes32 _release,
+		uint32 _forkBlock,
+		uint8 _track,
+		uint24 _semver,
+		bool _critical
+	)
+		public
+		onlyClientOwner
+	{
+		bytes32 newClient = clientOwner[msg.sender];
+		client[newClient].release[_release] = Release(
+			_forkBlock,
+			_track,
+			_semver,
+			_critical
+		);
 		client[newClient].current[_track] = _release;
-		ReleaseAdded(newClient, _forkBlock, _release, _track, _semver, _critical);
+		emit ReleaseAdded(
+			newClient,
+			_forkBlock,
+			_release,
+			_track,
+			_semver,
+			_critical
+		);
 	}
 
-	function addChecksum(bytes32 _release, bytes32 _platform, bytes32 _checksum) public only_client_owner {
-		var newClient = clientOwner[msg.sender];
+	/// Adds a `_checksum` for a `_release` binary targeting a given `_platform`. When a checksum is
+	/// succesfully added, a `ChecksumAdded` event is emitted. The same checksum can be added
+	/// multiple times (with different parameters), therefore the `ChecksumAdded` event can be
+	/// emitted multiple times for the same checksum.
+	function addChecksum(bytes32 _release, bytes32 _platform, bytes32 _checksum)
+		public
+		onlyClientOwner
+	{
+		bytes32 newClient = clientOwner[msg.sender];
+		require(releaseExists(newClient, _release));
 		client[newClient].build[_checksum] = Build(_release, _platform);
 		client[newClient].release[_release].checksum[_platform] = _checksum;
-		ChecksumAdded(newClient, _release, _platform, _checksum);
+		emit ChecksumAdded(
+			newClient,
+			_release,
+			_platform,
+			_checksum
+		);
 	}
 
 	// Admin functions
 
-	function addClient(bytes32 _client, address _owner) public only_owner {
-		client[_client].owner = _owner;
-		clientOwner[_owner] = _client;
-		ClientAdded(_client, _owner);
-	}
+	function setClient(bytes32 _client, address _owner)
+		public
+		onlyOwner
+		notClientOwner(_owner)
+	{
+		address owner = client[_client].owner;
```

Comment:
Same here, we should disallow setting `0x0` as an owner, the owner should just use `removeClient` instead.